File: mirai-core/src/main/java/net/mamoe/mirai/event/events/bot/BotEvent.java
package net.mamoe.mirai.event.events.bot;

import net.mamoe.mirai.Bot;
import net.mamoe.mirai.event.MiraiEvent;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public abstract class BotEvent extends MiraiEvent {
    private final Bot bot;

    public BotEvent(@NotNull Bot bot) {
        this.bot = Objects.requireNonNull(bot);
    }

    @NotNull
    public Bot getBot() {
        return bot;
    }

}

File: mirai-core/src/main/java/net/mamoe/mirai/event/events/bot/BotEvent.kt
package net.mamoe.mirai.event.events.bot

import net.mamoe.mirai.Bot
import net.mamoe.mirai.event.MiraiEvent

/**
 * @author Him188moe
 */
abstract class BotEvent(val bot: Bot) : MiraiEvent()

class BotLoginSucceedEvent(bot: Bot) : BotEvent(bot)